lib/point_quest/quests/commands/start_round.ex

```diff
@@ -0,0 +1,85 @@
+defmodule PointQuest.Quests.Commands.StartRound do
+  @moduledoc """
+  Command to start a round for the current quest.
+
+  Ensure that you're calling either `new/1` or `new!/1` followed by `execute/1` in order to
+  update the quest.
+  """
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  alias PointQuest.Quests
+  alias PointQuest.Authentication
+
+  @type t :: %__MODULE__{
+          quest_id: String.t()
+        }
+
+  @primary_key false
+  embedded_schema do
+    field :quest_id, :string
+  end
+
+  @spec new(map()) :: {:ok, t()}
+  @doc """
+  Creates a command to start a round on the current quest.
+
+  Returns a response tuple with the command.
+  """
+  def new(params) do
+    %__MODULE__{}
+    |> changeset(params)
+    |> apply_action(:update)
+  end
+
+  @spec new!(map()) :: t()
+  @doc """
+  Creates a command to start a round on the current quest.
+
+  Returns the command.
+  """
+  def new!(params) do
+    %__MODULE__{}
+    |> changeset(params)
+    |> apply_action!(:update)
+  end
+
+  @spec changeset(start_round :: t(), params :: map()) :: Ecto.Changeset.t(t())
+  @doc """
+  Creates a changeset from start_round and params.
+  """
+  def changeset(start_round, params \\ %{}) do
+    start_round
+    |> cast(params, [:quest_id])
+    |> validate_required([:quest_id])
+  end
+
+  defp repo(), do: Application.get_env(:point_quest, PointQuest.Behaviour.Quests.Repo)
+
+  @spec execute(start_round_command :: t(), actor :: Authentication.Actor.t()) ::
```

Comment:
`Authentication.Actor.PartyLeader.t()` right?

Reply:
I could go either way on this. At runtime we still have to check it so maybe the general type is the way to go. But also if they types could catch the issue it would be nice. FFTR

Reply:
You're absolutely right. I'll fix this in the morning and push up a change.